test/strings/types.jl

```diff
@@ -163,27 +163,29 @@ rs = RevString("foobar")
 @test rsplit(RevString("ailuj"),'l') == ["ju","ia"]
 @test parse(Float64,RevString("64")) === 46.0
 
-# reverseind
-for T in (String, UTF16String, UTF32String)
-    for prefix in ("", "abcd", "\U0001d6a4\U0001d4c1", "\U0001d6a4\U0001d4c1c", " \U0001d6a4\U0001d4c1")
-        for suffix in ("", "abcde", "\U0001d4c1β\U0001d6a4", "\U0001d4c1β\U0001d6a4c", " \U0001d4c1β\U0001d6a4")
-            for c in ('X', 'δ', '\U0001d6a5')
-                s = convert(T, string(prefix, c, suffix))
-                ri = search(reverse(s), c)
-                @test reverse(s) == RevString(s)
-                @test c == s[reverseind(s, ri)] == reverse(s)[ri]
-                s = RevString(s)
-                ri = search(reverse(s), c)
-                @test c == s[reverseind(s, ri)] == reverse(s)[ri]
-                s = convert(T, string(prefix, prefix, c, suffix, suffix))
-                pre = convert(T, prefix)
-                sb = SubString(s, nextind(pre, endof(pre)), endof(convert(T, string(prefix, prefix, c, suffix))))
-                ri = search(reverse(sb), c)
-                @test c == sb[reverseind(sb, ri)] == reverse(sb)[ri]
-            end
-        end
-    end
-end
+# # reverseind
+# for T in (String, GenericString)
```

Comment:
@stevengj I've tried fixing the tests using the method you pasted above, but there's still a failure with `SubString{GenericString}`. I've added a commit to the branch anyway, it would be great if you could have a look.

Here's a minimal reproducer:
```julia
s = SubString(GenericString("δ"), 1, 1)
reverseind(s, 1)
```